src/pybind/ceph_volume_client.py

```diff
@@ -1416,8 +1423,10 @@ def put_object(self, pool_name, object_name, data):
             raise CephFSVolumeClientError(msg)
         try:
             ioctx.write_full(object_name, data)
+            obj_version = ioctx.get_last_version()
```

Comment:
I thought I was explicit about how'd you want to do this atomically.
Pasting a part of my earlier comment.

"do compound write op to assert object version and then write to object using python bindings for 'rados_write_op_assert_version' followed by 'rados_write_op_write_full' [1]: put_object_with_assert_version(pool_name, object_name, data, version)

[1] See, https://github.com/ceph/ceph/blob/master/src/pybind/rados/rados.pyx#L1970
For usage example see, https://github.com/ceph/ceph/blob/master/src/test/pybind/test_rados.py#L509

Reply:
Here you're again checking version after writing to the object.

Reply:
You shouldn't write to the object until you're sure that the version of the object hasn't changed.

Reply:
> The other approach would be to do the object update (get data, process fetched data, put result) at a higher level (e.g. the manila cephfs driver). In this library provide interfaces to

Okay. I was under the impression that the paragraph above from your comment [1] allowed me a choice between the former (the paragraph above the one I quote) and the latter approach (the paragraph after the one I quote) and I chose former since I felt I was closer to it.

[1] https://github.com/ceph/ceph/pull/22455#discussion_r194521894

Reply:
I don't see python binding for `rados_write_op_assert_version` in rados.pyx or cephfs.pyx. Should I modify `write_full()` in rados.pyx [1] or create a new method solely for `rados_write_op_assert_version`?

[1] https://github.com/ceph/ceph/blob/master/src/pybind/rados/rados.pyx#L1970

Reply:
the latter, create pyhthon binding for rados_write_op_assert_version

Reply:
> Okay. I was under the impression that the paragraph above from your comment [1] allowed me a choice between the former (the paragraph above the one I quote) and the latter approach (the paragraph after the one I quote) and I chose former since I felt I was closer to it.

Irrespective of whether you implement a generic update_object() in this module, or expect atomic update_object() implemented at a higher level, you still have to implement `put_object_versioned()` and extend `get_object()` to also return object_version in this library.